packages/datetime/src/dateInput.tsx

```diff
@@ -423,6 +425,15 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         this.safeInvokeInputProp("onBlur", e);
     };
 
+    private handleInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.which === Keys.ENTER) {
+            const nextValue = this.createMoment(this.state.valueString);
+            const nextDate = fromMomentToDate(nextValue);
+            this.handleDateChange(nextDate, true);
+            this.inputRef.blur();
+        }
```

Comment:
Whoops, need to do `this.safeInvokeInputProp("onKeyDown", e);`.

Reply:
@cmslewis is this comment still relevant?